### ISSUE TITLE
Subscriptions don't have proper error handling

### ISSUE THREAD

**Issue by clayne11** [apollographql/apollo-client#3799]
**Issue**
Errors that happen during subscription set up don't get propagated back to `apollo-client` as errors but rather as successful calls.

**Intended outcome:**
Throwing an error while setting up a subscription should propagate a GraphQL error to the apollo client.

**Actual outcome:**
Throwing an error while setting up a subscription returns as a successful operation.

**Versions**
  System:
    OS: macOS High Sierra 10.13.6
  Binaries:
    Node: 8.11.3 - ~/.nvm/versions/node/v8.11.3/bin/node
    Yarn: 1.9.2 - /usr/local/bin/yarn
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.3/bin/npm
  Browsers:
    Chrome: 68.0.3440.106
    Firefox: 61.0.1
    Safari: 11.1.2
  npmPackages:
    "apollo-client": "^2.3.8"
    "apollo-server-express": "^2.0.0",

**Code**
If we look at the handling for mutations, we see that returned values from the server are [handled as `ApolloError`s:](https://github.com/apollographql/apollo-client/blob/d3c8d00e0ebecefd95a3e2da5bd4edb95d835941/packages/apollo-client/src/core/QueryManager.ts#L262-L303)

```tsx
      execute(this.link, operation).subscribe({
        next: (result: ExecutionResult) => {
          if (graphQLResultHasError(result) && errorPolicy === 'none') {
            error = new ApolloError({
              graphQLErrors: result.errors,
            });
            return;
          }


          this.mutationStore.markMutationResult(mutationId);


          if (fetchPolicy !== 'no-cache') {
            this.dataStore.markMutationResult({
              mutationId,
              result,
              document: mutation,
              variables: variables || {},
              updateQueries: generateUpdateQueriesInfo(),
              update: updateWithProxyFn,
            });
          }
          storeResult = result as FetchResult<T>;
        },


        error: (err: Error) => {
          this.mutationStore.markMutationError(mutationId, err);
          this.dataStore.markMutationComplete({
            mutationId,
            optimisticResponse,
          });
          this.broadcastQueries();


          this.setQuery(mutationId, () => ({ document: undefined }));
          reject(
            new ApolloError({
              networkError: err,
            }),
          );
        },


        complete: () => completeMutation().then(resolve, reject),
      });
```

As you can see, in the next block we check for `result.errors` and put the mutation into an error state if we see an error.

On subscriptions, however, there's [no handling at all for `result.errors`](https://github.com/apollographql/apollo-client/blob/d3c8d00e0ebecefd95a3e2da5bd4edb95d835941/packages/apollo-client/src/core/QueryManager.ts#L920-L950):

```tsx
      if (observers.length === 1) {
        const handler = {
          next: (result: FetchResult) => {
            if (isCacheEnabled) {
              this.dataStore.markSubscriptionResult(
                result,
                transformedDoc,
                variables,
              );
              this.broadcastQueries();
            }


            // It's slightly awkward that the data for subscriptions doesn't
            // come from the store.
            observers.forEach(obs => {
              // XXX I'd prefer a different way to handle errors for
              // subscriptions.
              if (obs.next) obs.next(result);
            });
          },
          error: (error: Error) => {
            observers.forEach(obs => {
              if (obs.error) obs.error(error);
            });
          },
        };


        // TODO: Should subscriptions also accept a `context` option to pass
        // through to links?
        const operation = this.buildOperationForLink(transformedDoc, variables);
        sub = execute(this.link, operation).subscribe(handler);
```

As far as I can tell this is completely broken.

**Comment by clayne11** [comment]
Fixed by #3800.

**Comment by wzup** [comment]
I just leave it here https://github.com/apollographql/apollo-client/issues/3858
Any chances to have normal error/warning stack trace in this repo?
